### DISCUSSION TITLE
Less strict detected fields

### DISCUSSION THREAD

**Discussion by loyd** [grafana/grafana#40645] [Dashboards, Panels & Visualizations]
The current version of `LOGFMT_REGEXP` restricts label's key to `([\w\(\)\[\]\{\}]+)`:
https://github.com/grafana/grafana/blob/70f68289f516e975795c0eec310e34e8171236c1/packages/grafana-data/src/utils/logs.ts#L11

In Rust main logging subsystem, [tracing](https://github.com/tokio-rs/tracing), it's ok to use `.` inside keys (for instance, `error.source = "..."`, `error.location = "..."`). Now such fields aren't detected by grafana, that's inconvenient.

Can the parser allow `.` inside keys or provide the way to alter the regexp in configuration (like with `Derived fields`)?

**Comment by loyd** [comment]
https://github.com/grafana/grafana/pull/48118

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
